From 5c09c87f219803bc1605b9b3a212fac186a2291a Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 09 Jan 2020 16:23:21 +0200
Subject: [PATCH] [SW-6739] arcboot: increse heap size

increase the heap size to improve support for synopsis
secureboot library as there are large mallocs, this was
as a suggetion from synopsis to improve run speed of
library.
stack was decresed because we have 16KB limited DCCM space

Change-Id: Id92b5fad1c3328f39c9b623c85d40fc977de326d
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/goya2_arc/arcboot_rom.lds b/goya2_arc/arcboot_rom.lds
index 4c9b062..3500e00 100644
--- a/goya2_arc/arcboot_rom.lds
+++ b/goya2_arc/arcboot_rom.lds
@@ -86,7 +86,7 @@
 	.heap : {
 		. = ALIGN(32 / 8);
 		__start_heap = . ;
-		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 4k)) ;
+		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 10k)) ;
 		__end_heap = . ;
 		. = ALIGN(0x8);
 	} > REGION_HEAP
@@ -94,7 +94,7 @@
 	.stack : {
 		. = ALIGN(32 / 8);
 		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 9k)) ;
+		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 3k)) ;
 		__stack_top = . ;
 	} > REGION_STACK
 }
\ No newline at end of file
